src/plugins/algorithms/als.cpp

```diff
@@ -63,9 +63,9 @@ namespace phylanx { namespace execution_tree { namespace primitives
         }
         auto regularization = arg2.scalar();
 
-        auto num_factors = extract_integer_value(args[2], name_, codename_);
+        auto num_factors = extract_integer_value(args[2], name_, codename_)[0];
```

Comment:
This would eb acandidate for the new function `extract_integer_scalar_value`, I think (same is true at several spots below).